dataset/dataset.go

```diff
@@ -777,6 +841,59 @@ func (c *Client) GetInstanceDimensions(ctx context.Context, serviceAuthToken, in
 	return
 }
 
+func (c *Client) GetInstanceDimensionsInBatches(ctx context.Context, userAuthToken, serviceAuthToken, instanceID string, batchSize, maxWorkers int) (dimensions Dimensions, err error) {
+
+	// Function to aggregate items.
+	// For the first received batch, as we have the total count information, will initialise the final structure of items with a fixed size equal to TotalCount.
+	// This serves two purposes:
+	//   - We can guarantee, even with concurrent calls, that values are returned in the same order that the API defines, by offsetting the index.
+	//   - We do a single memory allocation for the final array, making the code more memory efficient.
+	var processBatch InstanceDimensionsBatchProcessor = func(b Dimensions) (abort bool, err error) {
+		if len(dimensions.Items) == 0 { // first batch response being handled
+			dimensions.TotalCount = b.TotalCount
+			dimensions.Items = make([]Dimension, b.TotalCount)
+			dimensions.Count = b.TotalCount
+		}
+		for i := 0; i < len(b.Items); i++ {
+			dimensions.Items[i+b.Offset] = b.Items[i]
+		}
+		return false, nil
+	}
+
+	// call dataset API GetInstanceDimensions in batches and aggregate the responses
+	if err := c.GetInstanceDimensionsBatchProcess(ctx, userAuthToken, serviceAuthToken, instanceID, processBatch, batchSize, maxWorkers); err != nil {
+		return Dimensions{}, err
+	}
+
+	return dimensions, nil
+}
+
+// GetInstanceDimensionsBatchProcess gets the instance dimensions from the dataset API in batches, calling the provided function for each batch.
+func (c *Client) GetInstanceDimensionsBatchProcess(ctx context.Context, userAuthToken, serviceAuthToken, instanceID string, processBatch InstanceDimensionsBatchProcessor, batchSize, maxWorkers int) error {
+
+	// for each batch, obtain the dimensions starting at the provided offset, with a batch size limit,
+	// or the subste of IDs according to the provided offset, if a list of optionIDs was provided
```

Comment:
the second part of the comment can be removed (the subset of IDs is only for dimension options)